default_gateway.go

```diff
@@ -75,7 +76,7 @@ func (sb *sandbox) clearDefaultGW() error {
 		return nil
 	}
 
-	if ep == sb.getGatewayEndpoint() && !sb.inDelete {
+	if ep == sb.getGatewayEndpoint() && !sb.inDelete && len(sb.getConnectedEndpoints()) != 1 {
```

Comment:
We can use needDefaultGW() here; but it probably needs some changes.. If an endpoint is on an internal network it returns false. instead we have to continue and check other end-points as well.